src/dataflow/src/sink/tail.rs

```diff
@@ -62,11 +58,11 @@ pub fn tail<G>(
                 }
             }
 
-            // Sort results by time and convert to Vec<Row>. We use stable sort here even
-            // though it is slower because it will produce deterministic results since the
-            // cursor will always produce rows in the same order.
-            results.sort_by_key(|(time, _)| *time);
-            let mut results: Vec<Row> = results.into_iter().map(|(_, row)| row).collect();
+            // Sort results by time, which is the first column in the row. We
+            // use stable sort here even though it is slower because it will
+            // produce deterministic results since the cursor will always
+            // produce rows in the same order.
+            results.sort_by(|row1, row2| row1.unpack_first().cmp(&row2.unpack_first()));
```

Comment:
I'm not sure about this, it again seems to be mixing properties of the envelope and the connector - we now rely on the fact that the envelope stuffs the timestamp into the first column of the data.